File: napakalaki/src/Napakalaki/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Napakalaki;

import java.util.Random;

/**
 *
 * @author Marcos
 */
public class CultistPlayer extends Player {
    public static int totalCultistPlayers=0;
    public Cultist myCultistCard;

    public CultistPlayer(String _name) {
        super(_name);
    }
    
    public CultistPlayer(Player p, Cultist c){
        super(p);
        myCultistCard=c;
        totalCultistPlayers=totalCultistPlayers+1;
    }
    
    public Cultist getCultistPlayer(){
        return myCultistCard;
    }
    
    @Override
    protected int getCombatLevel(){
        int aux=super.getCombatLevel();
        aux=(int) (aux+(0.2*aux));
        aux=aux+(myCultistCard.getGainedLevels()*totalCultistPlayers);
        return aux;
    }
    
    @Override
    protected int getOponentLevel(Monster m){
        return m.getCombatLevelAgainstCultistPlayer();
    }
    
    @Override
    protected boolean shouldConvert(){
        return true;
    }
    
    @Override
    protected Treasure giveMeATreasure(){
        Random generator=new Random();
        int aleatorio=super.getVisibleTreasures().size();
        aleatorio=(generator.nextInt(aleatorio));
        return super.getVisibleTreasures().get(aleatorio);
    }
    
    @Override
    protected boolean canYouGiveMeATreasure(){
        if(enemy.getVisibleTreasures().isEmpty()){
            return false;
        }
        else{
            return true;
        }
    }
    
    public int getTotalCultistPlayers(){
        return totalCultistPlayers;
    }
    
}

File: napakalaki(ruby)/lib/cultist_player.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
require_relative 'player'
require_relative 'cultist'

class CultistPlayer < Player
  
  @@totalCultistPlayer=0
  
  def initialize (p,c)
    #Player.newplayer(p)
    super(p.name)
    @level=p.level
    @dead=p.dead
    @canISteal=p.canISteal
    @enemy=p.enemy
    @hiddenTreasures=p.hiddenTreasures
    @visibleTreasures=p.visibleTreasures
    @pendingBadConsequence=p.pendingBadConsequence
    @myCultistCard=c
    @@totalCultistPlayer+=1
  end
  
  
  def getCombatLevel
    nivel = super.getCombatLevel
    nivel += nivel*0.2
    nivel+= @myCultistCard.getGainedLevels
    nivel *= @@totalCultistPlayer
    return nivel
  end
  
  def getOponentLevel(m)
    return m.getCombatLevelAgainstCultistPlayer
  end
  
  def self.getTotalCultistPlayer
    return @@totalCultistPlayer
  end
  
  def shouldConvert
    return false
  end
  
  def giveMeATreasure
    aux= rand(super.VisibleTreasures.length)
    return super.VisibleTreasures.at(aux);
  end
  
  def canYouGiveMeATreasure
    if(super.enemy.visibleTreasures.empty?)
      return false
    else
      return true
    end
  end

  def to_s()
   " \n #{super.to_s} +  \n Carta de Cultist: #{@myCultistCard} \n Total de Jugadores Cultist: #{@@totalCultistPlayer}"
  end
  
  private :giveMeATreasure, :canYouGiveMeATreasure
end
end